DSL/com.larsreimann.safeds/model/SafeDS.ecore

```diff
@@ -194,6 +198,11 @@
         <eStructuralFeatures xsi:type="ecore:EReference" name="columnName" eType="#//SdsString" containment="true" />
         <eStructuralFeatures xsi:type="ecore:EReference" name="columnType" eType="#//SdsAbstractType" containment="true" />
     </eClassifiers>
+    <eClassifiers xsi:type="ecore:EClass" name="SdsSchemaEffectPredicate" eSuperTypes="#//SdsAbstractGoalCallable">
+        <eStructuralFeatures xsi:type="ecore:EAttribute" name="effect" eType="ecore:EDataType http://www.eclipse.org/emf/2002/Ecore#//EString" />
+        <eStructuralFeatures xsi:type="ecore:EReference" name="parameterList" eType="#//SdsParameterList" containment="true" />
+        <eStructuralFeatures xsi:type="ecore:EReference" name="resultList" eType="#//SdsResultList" containment="true" />
+    </eClassifiers>
```

Comment:
This does not appear in the grammar. What is its purpose?

Reply:
SdsSchemaEffectPrediate nodes are created for schema atomic effects, the only reason is to have a node with parameter list and result list just like predicate would have so the argument type checking and later result type inference can be done the same way as for predicates. Think of them  as built-in predicates that the user can not make.
see  staticAnalysis/schema/SchemaEffectPredicate.kt

Reply:
Wouldn't it make more sense to just create those classes by hand in Kotlin? Personally, I like to avoid Ecore as much as possible.

Reply:
Some more thoughts:

We can also add a validator just for `SdsSchemaEffectReference`, which checks the following:

- `SdsSchemaEffectReference` must be called
- `SdsSchemaEffectReference` must have m-n arguments (in case some are optional)
- `SdsSchemaEffectReference` must have arguments with the correct types

Reply:
I removed schemaEffectPredicate.

Currently, my understanding It that named type parameters won't work for the atomicSchemaEffects. is that correct?

Reply:
Can you give an example what you mean by "named type parameters"?

Reply:
sorry I meant named arguments. Instead of positional arguments, you can refer to the parameter name and pass arguments irrespective of the position in the paramterlist.

`$ReadSchema(datasetName="aName", datasetPath="aPath")`
and
`$ReadSchema(datasetPath="aPath", datasetName="aName")`
both are valid

Reply:
Would be good for consistency, if named arguments would work in this case.